### PR TITLE
Split ResourceManager implementations into own files + fix tests

### LINKED ISSUE #626: Split ResourceManager implementations to separate files
1. Split `ResourceManager` implementations (`LocalResourceManager`, `CCloudResourceManager`) to separate files.
2. Make base class `ResourceManager.getInstance()` be based on a dynamic backing Map instead of hardcoding:
   1. Make a `registerInstance()` method.
   2. Have extension (and perhaps test suite startup) register the two static instances.
  
Step two will be necessary to prevent circular dependencies, as well as paves the way for the more runtime-dynamic instances which will be used for direct / confluent platform connections.

### PR DESCRIPTION
**Pull request by jlrobins**
## Summary of Changes

- Split the two ResourceManager implementations into their own files, requiring:
  - Lots of imports changing.
  - Shift `ResourceManager.getInstance()`  to be dynamic registry based (was planned for supporting ad hoc direct connections, but mandatory now). This prevents circular imports between the module defining the base class, the registry, and those defining the implementations.
  - Fine-grained fix test suite that failed due to the resource loader instances not being registered.
- Closes #626 

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [ ] Added new
- [ ] Updated existing
- [ ] Deleted existing

##### Other

- [ ] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/storage/resourceLoaderInitialization.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Disposable } from "vscode";
+import { CCLOUD_CONNECTION_ID, LOCAL_CONNECTION_ID } from "../constants";
+import { CCloudResourceLoader } from "./ccloudResourceLoader";
+import { LocalResourceLoader } from "./localResourceLoader";
+import { ResourceLoader } from "./resourceLoader";
+
+/* This is used by both extension initialization and also the global test suite setup.
+
```
Comment (jlrobins) [inline src/storage/resourceLoaderInitialization.ts:8]: @alexeyraspopov Any idea why, when this function is defined in extension.ts, that it blows up the test suite a few seconds after when trying to run?

File: src/storage/resourceLoader.ts [hunk 2]
```diff
@@ -63,12 +41,18 @@ export abstract class ResourceLoader {
...
+
   /** Get the ResourceLoader subclass instance corresponding to the given connectionId */
   public static getInstance(connectionId: string): ResourceLoader {
-    if (connectionId === CCLOUD_CONNECTION_ID) {
-      return CCloudResourceLoader.getInstance();
-    } else if (connectionId === LOCAL_CONNECTION_ID) {
-      return LocalResourceLoader.getInstance();
+    const loader = ResourceLoader.registry.get(connectionId);
+    if (loader) {
+      return loader;
```
Comment (shouples) [inline src/storage/resourceLoader.ts:55]: 💯

**Review by shouples — APPROVED** [review]

**Review by noeldevelops — COMMENTED** [review]
do you think we'll build a new `storage/directResourceLoader.ts` or will it likely line up with local?

**Comment by jlrobins** [comment]
> do you think we'll build a new `storage/directResourceLoader.ts` or will it likely line up with local?

One or the other. I kinda think they'll end up being Real Close in any event, varying at most only by a little bit, in that they'll most likely not do any or much caching. CCLoudResourceLoadeer will remain the oddball one.

Am thinking:
  * DirectResourceLoader will not be a singleton, but instead constructed (and registered) on demand when user makes a direct connection. Its instance will know its connection id, and set it accordingly when interacting with sidecar.
  * Its code will probably be based off of LocalResourceLoader, but refactored a bit to know its connection id, and then whatever other changes needed. 8 ball is somewhat hazy.

**Pull request merged**